server/channels/app/platform/web_conn.go

```diff
@@ -789,6 +800,13 @@ func (wc *WebConn) ShouldSendEvent(msg *model.WebSocketEvent) bool {
 			wc.lastAllChannelMembersTime = 0
 		}
 
+		// Execute channel hook
+		if msg.GetBroadcast().ChannelHook != nil && msg.GetBroadcast().ChannelHook(wc.UserId, msg) {
+			// If hook returns true, that means message has been modified. We need
+			// to wipe off the pre-computed JSON
+			msg.RemovePrecomputedJSON()
```

Comment:
Nah, it just falls back to marshalling the JSON. It's a tradeoff. SInce we are modifying the message content in the hook, the existing precomputed JSON is obsolete. Therefore we need to marshal it from scratch again.